### PR TITLE
Move logging to go-kit logger

### PR DESCRIPTION
**Pull request by Gouthamve**
A cherry picked port of #3061

---
This change is [Reviewable](https://reviewable.io/reviews/prometheus/prometheus/3152)

### CODE REVIEW

**Comment by fabxc** [comment]
👍 fine from my side.

File: cmd/prometheus/main.go [hunk 1]
```diff
@@ -27,15 +27,20 @@ import (
 	"syscall"
 	"time"
 
+	k8s_runtime "k8s.io/apimachinery/pkg/util/runtime"
+
```
Comment (brian-brazil) [inline cmd/prometheus/main.go]: Why the blank line here?

File: cmd/prometheus/main.go [hunk 2]
```diff
@@ -203,13 +201,20 @@ func main() {
 
 	cfg.queryEngine.Timeout = time.Duration(cfg.queryTimeout)
 
-	logger := log.NewLogger(os.Stdout)
-	logger.SetLevel(cfg.logLevel)
-	logger.SetFormat(cfg.logFormat)
+	logger := promlog.New(cfg.logLevel)
+
+	// XXX(fabxc): The Kubernetes does background logging which we can only customize by modifying
```
Comment (brian-brazil) [inline cmd/prometheus/main.go]: s/The //

File: discovery/kubernetes/kubernetes.go [hunk 3]
```diff
@@ -171,7 +167,7 @@ func (d *Discovery) Run(ctx context.Context, ch chan<- []*config.TargetGroup) {
 			slw := cache.NewListWatchFromClient(rclient, "services", namespace, nil)
 			plw := cache.NewListWatchFromClient(rclient, "pods", namespace, nil)
 			eps := NewEndpoints(
-				d.logger.With("kubernetes_sd", "endpoint"),
+				log.With(d.logger, "k8s_sd", "endpoint"),
```
Comment (brian-brazil) [inline discovery/kubernetes/kubernetes.go]: This is role rather than the whole sd

File: promql/parse.go [hunk 4]
```diff
@@ -342,7 +342,7 @@ func (p *parser) recover(errp *error) {
 			buf := make([]byte, 64<<10)
 			buf = buf[:runtime.Stack(buf, false)]
 
-			log.Errorf("parser panic: %v\n%s", e, buf)
+			fmt.Fprintf(os.Stderr, "parser panic: %v\n%s", e, buf)
```
Comment (brian-brazil) [inline promql/parse.go]: This should go via logging rather than being sent to stderr

**Comment by Gouthamve** [comment]
@brian-brazil So the only thing left is the panic stack-trace. Can you take another look now?

**Comment by Gouthamve** [comment]
Rebased it against the current `dev-2.0`. As this is logging and touches every codepath, any minor change could potentially require a rebase.

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**